EIPS/eip-5528.md

```diff
@@ -0,0 +1,183 @@
+    ///   - escrow contract should update (Seller, Buyer) balance.
+    ///   - the seller should not call this function.
+    /// @param
+    ///   - to:
+    ///     In case of a buyer/seller contract, must be an escrow contract address.
+    ///     In case of an escrow contract, must be the user address who triggered this transaction.
+    ///   - _valuePayed: payable token amount
+    /// @return reason code. 0 is a success, otherwise is failure code.
+    function escrowRefund(address to, uint256 amount) public returns (uint32);
+
+    /// @notice withdraw token from the escrow account.
+    /// @dev
+    ///   - must be implemented in Escrow-Contract and optional for other contracts.
+    ///   - buyer is only available when escrow is successful, otherwise should call escrowRefund.
+    ///   - in case the escrow fails, the seller can refund the seller-token.
+    ///   - if the escrow is successful, the seller and buyers can get the exchanged tokens in their wallets.
+    /// @return reason code. 0 is success, otherwise is failure code.
+    function escrowWithdraw() public returns (uint32);
+
+}
+
+
+```
+
+## Rationale
+
+This standard proposes interfaces on top of the [EIP-20](./eip-20.md) standard.
+Each function should include constraint check logic.
+The escrow-contract should implement internal constraint logic such as
+ - Lock period
+ - Maximum(or minimum) number of investors
+ - Maximum(or minimum) number of tokens to fund
+ - Exchange rates of seller/buyer token
+ - KYC verification of users(It might require additional interface)
+ - etc
+
+The buyer-contract and seller-contract should not have constraint rules.
+
+Let's discuss the following functions.
+
+1. **constructor**
+
+An escrow contract will define success/failure conditions. It means constraint rules might not be changed forever (might be changed after being created for the market exchange rate.), so it guarantees escrow policy.
+
+2. **escrowFund**
+
+This function should run differently for buyers and sellers.
+
+2.1 [Seller]
+- The seller calls this function to be escrow-ready. The seller's token ownership(balance) will be transferred to the escrow contract and the escrow balance should be `(Seller: amount, Buyer: 0)`.
+- The seller can call this function multiple times depending on implementation, but preferred just one time.
+
+2.2 [Buyer]
+- During the escrow process, the buyer should be able to call this function to deposit funds into the escrow account.
+- The escrow balance should be  `(Seller: amount X exchange rate, Buyer: amount)`. The Buyer: the amount will be used for the refund process.
+- In a successful scenario, the seller's escrow balance should be `(Seller: -= amount X exchange rate, Buyer: += amount)`.
+
+3. **escrowRefund**
+
+This function should be invoked by buyers only.
+The buyer can call this function in the running state only. In the state of failure or success, could not be a success.
```

Comment:
Required: Not entirely sure what you mean here. Please make this less ambiguous.